lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -596,73 +600,67 @@ void mme_app_handle_initial_ue_message(mme_app_desc_t *mme_app_desc_p,
     is_guti_valid =
       mme_app_construct_guti(&plmn, &(initial_pP->opt_s_tmsi), &guti);
     if (is_guti_valid) {
-      ue_nas_ctx = emm_context_get_by_guti(&_emm_data, &guti);
-      if (ue_nas_ctx) {
-        // Get the UE context using mme_ue_s1ap_id
-        ue_context_p =
-          PARENT_STRUCT(ue_nas_ctx, struct ue_mm_context_s, emm_context);
-        DevAssert(ue_context_p != NULL);
-        if (ue_context_p != NULL) {
-          initial_pP->mme_ue_s1ap_id = ue_context_p->mme_ue_s1ap_id;
-          if (ue_context_p->enb_s1ap_id_key != INVALID_ENB_UE_S1AP_ID_KEY) {
-            /*
-             * Ideally this should never happen. When UE move to IDLE this key is set to INVALID.
-             * Note - This can happen if eNB detects RLF late and by that time UE sends Initial NAS message via new RRC
-             * connection
-             * However if this key is valid, remove the key from the hashtable.
-             */
+      ue_context_p =
+        mme_ue_context_exists_guti(&mme_app_desc_p->mme_ue_contexts, &guti);
+      if (ue_context_p) {
+        initial_pP->mme_ue_s1ap_id = ue_context_p->mme_ue_s1ap_id;
+        if (ue_context_p->enb_s1ap_id_key != INVALID_ENB_UE_S1AP_ID_KEY) {
+          /*
```

Comment:
Please re-format this comment to wrap text better

Reply:
Fixed in the updated commit